source/base/function_parser.cc

```diff
@@ -158,27 +158,25 @@ FunctionParser<dim>::init_muparser() const
   // check that we have not already initialized the parser on the
   // current thread, i.e., that the current function is only called
   // once per thread
-  Assert(fp.get().size() == 0, ExcInternalError());
+  ParserData &data = parser_data.get();
+  Assert(data.parsers.size() == 0 && data.vars.size() == 0, ExcInternalError());
 
-  // initialize the objects for the current thread (fp.get() and
-  // vars.get())
-  fp.get().reserve(this->n_components);
-  vars.get().resize(var_names.size());
+  // initialize the objects for the current thread
+  data.parsers.reserve(this->n_components);
+  data.vars.resize(var_names.size());
   for (unsigned int component = 0; component < this->n_components; ++component)
     {
-      fp.get().emplace_back(
-        std::make_unique<internal::FunctionParserImplementation::Parser>());
+      data.parsers.emplace_back(
+        new internal::FunctionParserImplementation::Parser());
       mu::Parser &parser =
         dynamic_cast<internal::FunctionParserImplementation::Parser &>(
-          *fp.get().back());
```

Comment:
@masterleinad this is the significant part - we used to call `ThreadLocalStorage::get()` all over the place. Each call requires locking and unlocking. We now call `get()` once at the top instead.